plugins/mongodb/README.md

```diff
@@ -0,0 +1,202 @@
+    delete_many({"salary": {$gt: $max}});
+ENDEMBED;
+delete_many(200000);
+```
+
+### Aggregation pipelines
+
+Aggregation pipelines offer a lot of functionality and there is way too much for me to show you here, but the plugin should work with any kind of aggregation pipeline that is passed in as long as it is valid bson. The aggregate command takes a single pipeline of the form [{doc}, ... ].
+
+
+```
+dataset({STRING first, STRING last, REAL salary}) aggregate(REAl max, REAL min, INTEGER order) := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll), batchSize(100))
+    aggregate([{$match: { salary: {$gte: $min, $lt: $max}}}, {$sort: {salary: $order}}]);
+ENDEMBED;
+aggregate(200000, 75000, 1);
+```
+In this example there are two stages in the pipeline. The first stage is a match operation and will find all the documents that have a salary greater than or equal to the min and less than the max. Then every document that gets returned from the first stage gets sorted by salary in the second stage. The order needs to be an integer and is 1 for ascending and -1 for descending.
+
+More information on and examples of MongoDB aggregation operations can be found here: [Manual](https://www.mongodb.com/docs/manual/aggregation/)
+
+### MongoDB runCommand function
+
+The runCommand function allows the user to run some commands at a database level instead of at the collection level. It has a single argument of a bson document. For each runCommand operation there is a unique return value. To get the data from the operation the record structure of the dataset must match the operation specific output of the value document.
+
+
+```
+dataset(layoutEmployee) findInfo(BOOLEAN mybool) := EMBED(mongodb : user(user), password(pwd), server(server), port(port), database(db),  collection('test1'))
+    runCommand(
+        {
+            findAndModify: "test1",
+            query: {first: "Anne", last: "Smith"},
+            remove: $mybool
+        }
+    );
+ENDEMBED;
+findInfo(true);
+```
+
+In this example the findAndModify command is ran on the test1 collection. Then it removes all the documents matching the query filter if remove is true.
+
+### Creating an Index
+
+Indexes can be useful for speeding up common queries on large collections that have similar fields.
+
+
+```
+boolean createIndex(INTEGER asc) := EMBED(mongodb : user(user), password(pwd), server(server), database('mydb'),  collection('test2'))
+    create_index({ "first": $asc, "last": $asc});
+ENDEMBED;
+createIndex(1);
+
+boolean createIndex(INTEGER asc, BOOLEAN myBool) := EMBED(mongodb : user(user), password(pwd), server(server), database('mydb'),  collection('test2'))
+    create_index({ "first": $asc, "last": $asc}, {name: "Person", unique: $myBool});
+ENDEMBED;
+createIndex(1, true);
+```
+
+In this example an index is created with the keys first and last and in ascending order.
+
```

Comment:
With each of the MongoDB examples above:  Does it make sense to link to MongoDB's documentation at any point?